content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-flows.md

```diff
@@ -322,15 +348,17 @@ The `CreateBoardingTicket` flow lets Mars Express self-issue a `BoardingTicket`
 Now that you've written the `CreateAndIssueMarsVoucher` flow, try writing the `CreateBoardingTicket` flow.
 
 You will need these variables:
+
 * `ticketDescription`
-* `daysUntilLaunch`
+* `launchDate`
 
 You must inject these services:
-* `JsonMarshallingService`
-* `FlowEngine`
-* `FlowIdentity`
-* `TransactionBuilderFactory`
-* `NotaryLookupService`
+
+* `FlowEngine`: Provides basic control over a flow it is injected into, such as calling subflows, performing asynchronous tasks and putting the flow to sleep for a period of time. This flow needs `FlowEngine` because you are utilizing subflows such as `FinalityFlow`.
+* `FlowIdentity`: Allows obtaining the identity of the node running the flow. You must inject this service because one of the output states needs its own identity.
```

Comment:
Same comment about starting some of these with "allows". It doesn't read well...

Reply:
changed